core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -2087,6 +2087,128 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
+
+    val config = createConfig
+    client = Admin.create(config)
+    try {
+      client.createTopics(Collections.singleton(
+        new NewTopic(testTopicName, 1, 1.toShort)
+      )).all().get()
+      waitForTopics(client, List(testTopicName), List())
+
+      classicGroup.subscribe(Collections.singleton(testTopicName))
+      classicGroup.poll(JDuration.ofMillis(1000))
+      consumerGroup.subscribe(Collections.singleton(testTopicName))
+      consumerGroup.poll(JDuration.ofMillis(1000))
+      shareGroup.subscribe(Collections.singleton(testTopicName))
+      shareGroup.poll(JDuration.ofMillis(1000))
+
+      TestUtils.waitUntilTrue(() => {
+        val groups = client.listGroups().all().get()
+        groups.size() == 3
+      }, "Expected to find all groups")
+
+      val classicGroupListing = new GroupListing(classicGroupId, Optional.of(GroupType.CLASSIC), "consumer")
+      val consumerGroupListing = new GroupListing(consumerGroupId, Optional.of(GroupType.CONSUMER), "consumer")
+      val shareGroupListing = new GroupListing(shareGroupId, Optional.of(GroupType.SHARE), "share")
+
+      var listGroupsResult = client.listGroups()
+      assertTrue(listGroupsResult.errors().get().isEmpty)
+      assertEquals(Set(classicGroupListing, consumerGroupListing, shareGroupListing), listGroupsResult.all().get().asScala.toSet)
+      assertEquals(Set(classicGroupListing, consumerGroupListing, shareGroupListing), listGroupsResult.valid().get().asScala.toSet)
+
+      listGroupsResult = client.listGroups(new ListGroupsOptions().withTypes(java.util.Set.of(GroupType.CLASSIC)))
+      assertTrue(listGroupsResult.errors().get().isEmpty)
+      assertEquals(Set(classicGroupListing), listGroupsResult.all().get().asScala.toSet)
+      assertEquals(Set(classicGroupListing), listGroupsResult.valid().get().asScala.toSet)
+
+      listGroupsResult = client.listGroups(new ListGroupsOptions().withTypes(java.util.Set.of(GroupType.CONSUMER)))
+      assertTrue(listGroupsResult.errors().get().isEmpty)
+      assertEquals(Set(consumerGroupListing), listGroupsResult.all().get().asScala.toSet)
+      assertEquals(Set(consumerGroupListing), listGroupsResult.valid().get().asScala.toSet)
+
+      listGroupsResult = client.listGroups(new ListGroupsOptions().withTypes(java.util.Set.of(GroupType.SHARE)))
+      assertTrue(listGroupsResult.errors().get().isEmpty)
+      assertEquals(Set(shareGroupListing), listGroupsResult.all().get().asScala.toSet)
+      assertEquals(Set(shareGroupListing), listGroupsResult.valid().get().asScala.toSet)
+    } finally {
+      Utils.closeQuietly(classicGroup, "classicGroup")
+      Utils.closeQuietly(consumerGroup, "consumerGroup")
+      Utils.closeQuietly(shareGroup, "shareGroup")
+      Utils.closeQuietly(client, "adminClient")
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = Array("kraft"))
+  def testDescribeClassicGroups(quorum: String): Unit = {
```

Comment:
This test doesn't really have enough variation. You could make a simple consumer group by using the admin client to set a committed offset without actually subscribing. It would be nice to make a connect group, but that's more work, so I think that just the simple consumer group would be enough.

Reply:
@FrankYang0529, please use `Admin#alterConsumerGroupOffsets` to set a committed offset. The coordinator will create a classic group with an empty protocol, known as a simple consumer group, for you.

Reply:
Hi @AndrewJSchofield / @chia7712, thanks for your suggestion. I add simple consumer group to all test cases. Could you review again when you have time? Thanks.